riscv-pages/src/page.rs

```diff
@@ -38,6 +38,22 @@ impl PageSize {
     pub const fn num_4k_pages(val: u64) -> u64 {
         (val + PageSize::Size4k as u64 - 1) / (PageSize::Size4k as u64)
     }
+    /// Return the number of bits to shift to get one page.
+    /// i.e. 4096 == 2^12 == 1 << 12
+    pub fn page_shift(&self) -> u64 {
```

Comment:
This duplicates some code in PageTable::addr_shift, would be good to have that function use this if possible.